language/move-binary-format/src/compatibility.rs

```diff
@@ -156,11 +156,19 @@ impl Compatibility {
             {
                 // if it was public(script), it must remain public(script)
                 // if it was not public(script), it _cannot_ become public(script)
-                old_func.is_entry == new_func.is_entry
+                // if it was private before, it can become public(script)
+                old_func.visibility == Visibility::Private || old_func.is_entry == new_func.is_entry
```

Comment:
Not sure if we can/should change this behaviour as it's already baked into bytecode format 5, cc @tnowacki for thoughts.

Reply:
Discussed this offline -- this change is actually necessary to maintain backwards compatibility because of the change from iterating over only `exposed` functions to iterating over all functions.